programmers-coding-test/wonji/greedy/숫자카드게임.js

```diff
@@ -0,0 +1,18 @@
+let N = prompt(`행의 개수 입력`);
+let M = prompt(`열의 개수 입력`);
+let data = prompt(`숫자들 입력`);
+
+function solution(data,N,M){
+    let result = 0;
+    let min_value = 0;
+    
+    for(let i = 1; i < N; i++) {
+        for (let j = 1; j < M; j++) {
+            min_value = Math.min(data[j]);
+            result = Math.max(result, min_value);
+        }
+    }
```

Comment:
Iterable 객체(배열, 문자열, 오브젝트 등의 반복 객체들)에 접근할 때 첫번째 인덱스는 1이 아닌 0이다.
1차원 배열에 접근하여 최소값을 반환하면 되므로 2차원 주소까지 접근할 필요가 없다.
따라서 2중 반복문이 아닌 단일 반복으로 해결할 수 있다.

```suggestion
    for (let i = 0; i < N; i++) {
        min_value = Math.min(...data[i]); // data[i] 의 배열을 “...”을 통해 언패킹한 뒤 최솟값을 반환한다.
        result = Math.max(result, min_value);
    }
```

아래는 숫자 하나의 최소값을 의미함.
```javascript
Math.min(data[j]);
```
즉, 아래와 같은 것임. 심지어 문자를 담아줬으므로 문자코드(아스키코드)의 최소값을 반환하게 됨.
따라서 정수형으로 변환하는 과정이 필요함.
```javascript
Math.min(“3”);
```

Reply:
Iterable 객체의 map 메서드를 활용하면 한 줄로도 풀 수 있다.
```suggestion
    result = Math.max(...cards.map(card => Math.min(...card)));
```